Content.Server/Database/ServerDbManager.cs

```diff
@@ -86,131 +48,360 @@ public void Init()
+                // By returning null, this should cause re-initialization which will fix the situation.
+                Logger.WarningS("c.s.db.serverdbmanager", "Had to reinitialize preferences[{0}] (user {1}) because it existed but had no profiles.", prefs.Id, userId);
+                return null;
+            }
+
+            var selected = await ServerDbContext.PreferenceProfiles
+                .Where(p => p.PreferenceId == prefs.Id)
+                .Select(p => p.Profile.Slot)
+                .SingleOrDefaultAsync();
+
+            var maxSlot = prefs.Profiles.Max(p => p.Slot) + 1;
+            var profiles = new Dictionary<int, ICharacterProfile>(maxSlot);
+            foreach (var profile in prefs.Profiles)
+            {
+                profiles[profile.Slot] = profile.ConvertProfile();
+            }
+
+            if (!profiles.ContainsKey(selected))
+                selected = profiles.First().Key;
+
+            return new PlayerPreferences(profiles, selected);
         }
 
-        public Task AssignUserIdAsync(string name, NetUserId userId)
+        public async Task AssignUserIdAsync(string name, NetUserId userId)
         {
-            return _db.AssignUserIdAsync(name, userId);
+            ServerDbContext.AssignedUsers.Add(new AssignedUser
+            {
+                UserId = userId,
+                UserName = name
+            });
+
+            await ServerDbContext.SaveChangesAsync();
         }
 
-        public Task<NetUserId?> GetAssignedUserIdAsync(string name)
+        public async Task<NetUserId?> GetAssignedUserIdAsync(string name)
         {
-            return _db.GetAssignedUserIdAsync(name);
+            var assigned = await ServerDbContext.AssignedUsers.SingleOrDefaultAsync(p => p.UserName == name);
+            return (NetUserId?) assigned?.UserId;
         }
 
-        public Task<ServerBanDef?> GetServerBanAsync(IPAddress? address, NetUserId? userId)
+        public async Task<ServerBan?> GetServerBanAsync(IPAddress? address, NetUserId? userId)
         {
-            return _db.GetServerBanAsync(address, userId);
+            // SQLite can't do the net masking stuff we need to match IP address ranges.
+            // So just pull down the whole list of IP bans into memory too.
+            var bans = await ServerDbContext.Bans
+                //.Where(p => p.UserId == userId || p.UserId == null)
+                .Include(p => p.Unban)
+                .Where(p => p.Unban == null && (p.ExpirationTime == null || p.ExpirationTime.Value > DateTime.UtcNow))
+                .ToListAsync();
+
+            foreach (var ban in bans)
+            {
+                if (address is {} && ban.Address is {} && address.IsInSubnet(ban.Address.Value.Item1, ban.Address.Value.Item2))
+                {
+                    return ban;
+                }
+
+                if (userId is {} id && ban.UserId == id.UserId)
+                {
+                    return ban;
+                }
+            }
```

Comment:
Making this slow in-memory path the default *for postgres as well* is an ABSOLUTE NO-NO.

Reply:
Ah, forgot to fix that one up with squishing problems all over the place with it. Will make it construct a SQLite specific query then, and do better one for all others.